django_gamification/models.py

```diff
@@ -60,6 +60,7 @@ def save(self, *args, **kwargs):
                     description=self.description,
                     progression=Progression.objects.create(target=self.progression_target) if self.progression_target
                     else None,
+                    next_badge=self.next_badge,
```

Comment:
This is the part that looks like it is the trickiest to implement. The value for `next_badge` needs to be an instance of `Badge` but in this context you're assigning a `BadgeDefinition`. Perhaps the best way to accomplish this is to make all the `Badge`s we need to (without `next_badge` set) and then populate that field afterwards in a separate step?

Reply:
Ah, right.  That makes sense; thanks for clarifying.  Just finished editing that part now and extended the `test_models.py`.  I'm trying to figure out how to get the changes I made in my fork's branch to go into this PR now... haha.  (I normally don't use forks, so this is a first for me.)  :)

Reply:
Ah, nevermind, it's there after all (https://github.com/mattjegan/django-gamification/pull/11/commits) but just not showing in this main view... (at least for me)